go/plugins/compat_oai/compat_oai.go

```diff
@@ -0,0 +1,194 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package compat_oai
+
+import (
+	"context"
+	"strings"
+	"sync"
+
+	"github.com/firebase/genkit/go/ai"
+	"github.com/firebase/genkit/go/genkit"
+	openaiGo "github.com/openai/openai-go"
+	"github.com/openai/openai-go/option"
+)
+
+var (
+	// BasicText describes model capabilities for text-only GPT models.
+	BasicText = ai.ModelInfo{
+		Supports: &ai.ModelSupports{
+			Multiturn:  true,
+			Tools:      true,
+			SystemRole: true,
+			Media:      false,
+		},
+	}
+
+	// Multimodal describes model capabilities for multimodal GPT models.
+	Multimodal = ai.ModelInfo{
+		Supports: &ai.ModelSupports{
+			Multiturn:  true,
+			Tools:      true,
+			SystemRole: true,
+			Media:      true,
+		},
+	}
+)
+
+type OpenAICompatible struct {
+	mu       sync.Mutex
+	initted  bool
+	client   *openaiGo.Client
+	Opts     []option.RequestOption
+	Provider string
```

Comment:
Some comments here would be good. What is provider?

Reply:
added here: https://github.com/kekoawong/genkit/pull/11

Reply:
Hmm, don't see anything here.

Reply:
I added a small comment here: https://github.com/kekoawong/genkit/blob/efc737682364ddb300b99259ac0b0a7bab7dae47/go/plugins/compat_oai/compat_oai.go#L52 I can add more info

Reply:
We'd like comments on each field of `OpenAICompatible`. As a user, how do I know what you put for `Provider`? Comment explaining what it is and examples of what goes in there is helpful without looking up documentation.

Reply:
added here: https://github.com/kekoawong/genkit/pull/13/files